.github/workflows/lint.yml

```
@@ -11,35 +11,31 @@
+        uses: actions/setup-python@v5
+        id: cpython3
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+          cache-dependency-path: requirements.txt
+          allow-prereleases: true
+      - run: pip install -r requirements.txt
+      - run: pip install black ruff pytest pytest-cov
+      - run: ruff check .
+      - name: Run mypy if on 3.12, pytype otherwise
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2193:warning:1:54: The arguments to this comparison can never be equal. Make sure your syntax is correct

(shellcheck)